scripts/pathfinders_validation.py

```diff
@@ -0,0 +1,439 @@
+        ),
+        "Project finance changes": ta.TableSchema(
+            id_tag="PF-TABLE-ID-22",
+            worksheet_name="Finances",
+            section="Portfolio level finances",
+            columns={
+                "Change Number": pa.Column(int, unique=True),  # TODO: confirm what this is?
+                # TODO: allowed projects in the dropdown are unknown at this stage, does this need to be validated?
+                "Project Funding Moved From": pa.Column(str),
+                "Intervention Theme Moved From": pa.Column(str),  # TODO: isin
+                "Project Funding Moved To": pa.Column(str),
+                "Intervention Theme Moved To": pa.Column(str),  # TODO: isin
+                "Amount Moved": pa.Column(float, checks=[pa.Check.greater_than(0), pa.Check.less_than(5000000)]),
+                "Changes Made (100 words Max)": pa.Column(str, pa.Check.word_count_less_than(100)),
+                "Reason for Change  (100 words Max)": pa.Column(str, pa.Check.word_count_less_than(100)),
+                "Forecast or Actual Change": pa.Column(str, pa.Check.isin(PFEnums.FORECAST_ACTUAL)),
+                "Reporting Period Change Took Place": pa.Column(str),  # TODO: isin
+            },
+            messages={
+                **messages,
+                "greater_than": "Amount moved must be greater than 0.",
+                "less_than": "Amount moved must be less than 5m.",
+                "word_count_less_than": "Please enter a maximum of 100 words.",
+            },
+            # 8 merged columns
+            drop_empty_rows=True,
+        ),
+        "Sign off name": ta.TableSchema(
+            id_tag="PF-TABLE-ID-23",
+            worksheet_name="Sign off",
+            section="Sign off",
+            columns={
+                "Name": pa.Column(str),
+            },
+            messages=messages,
+        ),
+        "Sign off role": ta.TableSchema(
+            id_tag="PF-TABLE-ID-24",
+            worksheet_name="Sign off",
+            section="Sign off",
+            columns={
+                "Role": pa.Column(str),
+            },
+            messages=messages,
+        ),
+        "Sign off date": ta.TableSchema(
+            id_tag="PF-TABLE-ID-25",
+            worksheet_name="Sign off",
+            section="Sign off",
+            columns={
+                "Date": pa.Column(datetime),
+            },
+            messages={**messages, "coerce_dtype": "You entered text instead of a date."},
+            row_idxs_to_drop=[0],
+        ),
+    }
+
+
+def run_pathfinders_schema_validation():
+    workbook = pd.read_excel(
+        resources / "pathfinders-validation-example-spreadsheet.xlsx", index_col=None, header=None, sheet_name=None
+    )
+
+    pf_table_schemas = get_pathfinders_table_schemas()
+
+    output_tables = defaultdict(list)
+    final_errors = []
+    for table_name, schema in pf_table_schemas.items():
+        extracted_tables = schema.extract(workbook[schema.worksheet_name])
```

Comment:
I originally implemented `extract` like this but then thought it was better to slim down the interface to just expect the worksheet. Being coupled to the full workbook felt unnecessary and a bit misleading because `extract` won't work across multiple sheets. It also made the testing feel a bit clunky as you had to put each mock `df` into a `dict`. I'm a bit torn on what's best, can see benefits in both.